github/users_emails.go

```diff
@@ -70,3 +70,22 @@ func (s *UsersService) DeleteEmails(ctx context.Context, emails []string) (*Resp
 
 	return s.client.Do(ctx, req, nil)
 }
+
+// SetEmailVisibility sets the visibility for the primary email address of the authenticated user.
+//
+// GitHub API docs: https://docs.github.com/en/rest/users/emails#set-primary-email-visibility-for-the-authenticated-user
+func (s *UsersService) SetEmailVisibility(ctx context.Context, updateVisiblilityReq *UserEmail) ([]*UserEmail, *Response, error) {
```

Comment:
This endpoint only accepts one of the 4 fields of `UserEmail`: `visibility`.
As such, it can be confusing to the user to have to build a `UserEmail` struct for this API call.
Instead, let's please do the following:

```suggestion
// SetEmailVisibility sets the visibility for the primary email address of the authenticated user.
// `visibility` can be "private" or "public".
//
// GitHub API docs: https://docs.github.com/en/rest/users/emails#set-primary-email-visibility-for-the-authenticated-user
func (s *UsersService) SetEmailVisibility(ctx context.Context, visibility string) ([]*UserEmail, *Response, error) {
	updateVisiblilityReq := &UserEmail{Visibility: &visibility}
```

Reply:
I have made the suggested changes @gmlewis